internal/security/setup/option/generate.go

```diff
@@ -36,17 +36,24 @@ func Generate() func(*PkiInitOption) (exitCode, error) {
 		if statusCode, err := generatePkis(); err != nil {
 			return statusCode, err
 		}
-		generatedDirPath := filepath.Join(getXdgRuntimeDir(), pkiInitGeneratedDir)
+		generatedDirPath := filepath.Join(getWorkDir(), pkiInitGeneratedDir)
 		// Shred the CA private key before deploy
 		caPrivateKeyFile := filepath.Join(generatedDirPath, caServiceName, tlsSecretFileName)
 		if err := secureEraseFile(caPrivateKeyFile); err != nil {
 			return exitWithError, err
 		}
 
-		if err := deploy(generatedDirPath, pkiInitDeployDir); err != nil {
+		deployDir, err := getDeployDir()
+		if err != nil {
 			return exitWithError, err
 		}
 
+		if err := deploy(generatedDirPath, deployDir); err != nil {
+			return exitWithError, err
+		}
+
+		os.RemoveAll(generatedDirPath)
```

Comment:
make this as `defer`, so that it will be cleaned up whether there is error or not.